### PR TITLE
[LTX2]  better timing and profiling capabilities

### PR DESCRIPTION
**Pull request by mbohlool**
Description:
This PR introduces better timing and profiling capabilities to the LTX2 generation pipeline to help identify performance bottlenecks.

Key Changes:

Detailed Timing: Added time.perf_counter() blocks and jax.block_until_ready() calls across the pipeline to accurately measure text encoding, connector passes, denoising steps, VAE decoding, and post-processing.

Multi-Pass Execution: Updated generate_ltx2.py to support a three-stage execution flow:

Warmup Pass: For JIT compilation.

Generation Pass: For actual output and standard timing.

Profiling Pass: (Optional) Captured via max_utils.Profiler for a subset of steps.

Enhanced Logging: Added a summary table for Load, Compile, and Inference times.

e.g.

```
==================================================
  TIMING SUMMARY
==================================================
  Load (checkpoint):     106.7s
  Compile:                81.5s
  ────────────────────────────────────────
  Inference:              16.4s
    Text Encoding:         3.6s
    Preparation:           0.0s
    Connectors:            0.0s
    Denoising:             7.2s
    Latent Upsampler:      0.0s
    Latent Processing:     0.6s
    Video VAE:             1.5s
    Video Post:            3.2s
    Audio VAE:             0.1s
    Vocoder:               0.2s
==================================================
```

Config Updates: Added skip_first_n_steps_for_profiler and profiler_steps to the LTX2 configuration.

Memory Management: Explicitly deletes large tensors (out, videos, audios) before the profiling run to prevent OOM.

### CODE REVIEW

**Comment by Perseus14** [comment]
@mbohlool Could you add a table with the latency gain (single video and amortized throughput) of this change with the baseline (main)?

Thanks!

**Comment by mbohlool** [comment]
@Perseus14 change the PR to focus only on the timing and profiling part. I explored the performance tweaking later. PTAL.

File: src/maxdiffusion/pipelines/ltx2/ltx2_pipeline.py [hunk 1]
```diff
@@ -1350,10 +1360,11 @@ def __call__(
         activation_axes = nn.logical_to_mesh_axes(("activation_batch", "activation_length", "activation_embed"))
         spec = NamedSharding(self.mesh, P(*activation_axes))
         video_embeds_sharded = jax.device_put(video_embeds, spec)
-        audio_embeds_sharded = jax.device_put(audio_embeds, spec)
+        audio_embeds_sharded = audio_embeds
```
Comment (Perseus14) [inline src/maxdiffusion/pipelines/ltx2/ltx2_pipeline.py]: @prishajain1 Could you check whether this will cause issues?

File: src/maxdiffusion/generate_ltx2.py [hunk 2]
```diff
@@ -210,29 +244,51 @@ def run(config, pipeline=None, filename_prefix="", commit_hash=None):
...
-    else:
-      max_logging.log("Warning: Number of videos is zero, cannot calculate generation_time_per_video.")
+  max_logging.log(
+      f"\n{'=' * 50}\n"
+      f"  TIMING SUMMARY\n"
+      f"{'=' * 50}\n"
+      f"  Load (checkpoint):   {load_time:>7.1f}s\n"
+      f"  Compile:             {compile_time:>7.1f}s\n"
+      f"  {'─' * 40}\n"
+      f"  Inference:           {generation_time:>7.1f}s\n"
```
Comment (Perseus14) [inline src/maxdiffusion/generate_ltx2.py:254]: Is it possible to print a component wise split here for quick analysis now that we are timing all the components?

Elisa has done something like this for the WAN pipelines [here](https://github.com/AI-Hypercomputer/maxdiffusion/pull/392/changes#diff-0a5dc65afc89ff23b646c90ee86adb3309cef9f1b3823677eaeaa52bcd2ee794)

File: src/maxdiffusion/pipelines/ltx2/ltx2_pipeline.py (hunk 1 above)
Comment (mbohlool) [reply src/maxdiffusion/pipelines/ltx2/ltx2_pipeline.py]: reverted

File: src/maxdiffusion/generate_ltx2.py (hunk 2 above)
Comment (mbohlool) [reply src/maxdiffusion/generate_ltx2.py:254]: done. check PR description for an example.

**Review by Perseus14 — APPROVED** [review]
